torch/_inductor/lowering.py

```diff
@@ -6223,6 +6224,93 @@ def while_loop(cond_fn, body_fn, carried_inputs, additional_inputs):
     return list(map(TensorBox.create, result))
 
 
+@register_lowering(scan_op)
+def scan(combine_subgraph, init, inputs, dim, reverse, additional_inputs):
+    from torch._higher_order_ops.scan import _extract_carry_and_out, stack_y
+
+    num_init_leaves = len(init)
+    if any(map(is_triton, [init, inputs, *additional_inputs])):
+        msg = "control flow operator: torch.scan."
+        if stack_trace := V.graph.current_node.meta.get("stack_trace", None):
+            msg = f"{msg} Found from : \n {stack_trace}"
+        V.graph.disable_cudagraphs_reason = msg
+
+    def extract_scan_args(combine_subraph, init, xs, dim, reverse, additional_inputs):
+        return combine_subraph, init, xs, dim, reverse, additional_inputs
+
+    def _to_out_variant_graph(
```

Comment:
This isn't the right way to convert this to out variants (if I understand what this code is doing correctly).
 
The right way is to modify the layouts of the inner subgraph in a similar way to `concatkernel`: https://github.com/pytorch/pytorch/blob/31c4e0d37d8efc37a0697159e5b9121ec34d5141/torch/_inductor/ir.py#L4183

Reply:
As discussed offline uses view + copy_ to handle this. We create a symint that's used in subgraph for the slice's strides and that symint is set to the current iteration counter before ops in subgraph runs.

Reply:
Please document what this rewrite does. Specifically, describe what the calling convention looks like at the beginning and what it looks like after.